### DISCUSSION TITLE
I'm really new to neovim treesitter, Is there any way or any command (like :LspAttach) to load a certain parser on a buf I want?

### DISCUSSION THREAD

**Discussion by G4RASU** [nvim-treesitter/nvim-treesitter#5140] [Q&A] [answered]
GIven a example, I want edit or browser a temp json string I paste into a unnamed buf. Is there a way to use json parser in this buf?

**Comment by amaanq** [comment]
`:set ft=json`

**Reply by G4RASU** [reply]
Thanks a lot. I tried this before and I cannot get any node from telescope treesitter picker although the highlight is working. I thought this is  because the parse is not load, but now it seems to be a another problem.

**Comment by clason** [comment, marked as answer]
`:h vim.treesitter.start()` (this sort of thing is handled by Neovim, so you should first read its documentation).

**Reply by G4RASU** [reply]
Also thanks! This function not only works but also avoid changing filetype!

**Comment by G4RASU** [comment]
In conclusion, there are 2 ways load treesitter to a buf:
1. by setting the corresponding filetype `set fy=json`
2. by calling function `vim.treesitter.start(bufnr, lang)`

**Reply by clason** [reply]
No. These two do different things. I would really recommend reading the documentation to understand what is happening.

**Reply by G4RASU** [reply]
Okey, got it! Thank you for your patience.
